### PR TITLE
Deprecate usage of undefined blocks

### PR DESCRIPTION
**Pull request by julienfalque**
Alternative to #2277.

### CODE REVIEW

File: test/Twig/Tests/Fixtures/functions/recursive_block_with_inheritance.test [hunk 1]
```diff
@@ -12,7 +12,7 @@
 {% block list %}<ul>{{ block('children') }}</ul>{% endblock %}
 {% block children %}{% set currentItem = item %}{% for item in currentItem %}{{ block('item') }}{% endfor %}{% set item = currentItem %}{% endblock %}
 {% block item %}<li>{% if item is not iterable %}{{ block('label') }}{% else %}{{ block('list') }}{% endif %}</li>{% endblock %}
-{% block label %}{{ item }}{{ block('unknown') }}{% endblock %}
```
Comment (stof) [inline test/Twig/Tests/Fixtures/functions/recursive_block_with_inheritance.test]: the case of an unknown block must be kept in a separate test marked as legacy, to ensure we don't break its support in 1.x

File: test/Twig/Tests/TemplateTest.php [hunk 2]
```diff
@@ -234,6 +236,21 @@ public function testGetAttributeWithTemplateAsObjectForDeprecations()
     }
 
     /**
+     * @group legacy
+     * @expectedDeprecation Displaying undefined block "unknown" in template "index.twig" is deprecated since version 1.29 and will throw an exception in 2.0.
+     * @expectedDeprecation Displaying undefined block "unknown" in template "index.twig" is deprecated since version 1.29 and will throw an exception in 2.0.
+     */
+    public function testRenderBlockWithUndefinedBlock()
+    {
+        // to be removed in 2.0
```
Comment (stof) [inline test/Twig/Tests/TemplateTest.php]: this comment must be removed. Tests to be removed in 2.0 are all the legacy tests of 1.x, so there is not need to add this comment

File: test/Twig/Tests/Fixtures/functions/recursive_block_with_inheritance.test (hunk 1 above)
Comment (julienfalque) [inline test/Twig/Tests/Fixtures/functions/recursive_block_with_inheritance.test]: I added a legacy integration test.

File: test/Twig/Tests/TemplateTest.php (hunk 2 above)
Comment (julienfalque) [inline test/Twig/Tests/TemplateTest.php]: Comment removed.

**Comment by fabpot** [comment]
Thank you @julienfalque.

**Pull request merged**